### PR TITLE
soundio: rename SoundDeviceError > SoundDeviceStatus

### PR DESCRIPTION
**Pull request by ronso0**
I found this a bit misleading/annoying that a device status is named "Error", as well as the inadequate names `SOUNDDEVICE_ERROR_OK` and `SOUNDDEVICE_ERROR_ERR`

So now it's
``` c++
enum SoundDeviceStatus {
    SOUNDDEVICE_ERROR = -1,
    SOUNDDEVICE_OK = 0,
    SOUNDDEVICE_ERROR_DUPLICATE_OUTPUT_CHANNEL,
    SOUNDDEVICE_ERROR_EXCESSIVE_OUTPUT_CHANNEL,
    SOUNDDEVICE_ERROR_EXCESSIVE_INPUT_CHANNEL,
    SOUNDDEVICE_ERROR_DEVICE_COUNT
};
```

### CODE REVIEW

File: src/soundio/soundmanager.cpp [hunk 1]
```diff
@@ -562,11 +562,11 @@ SoundDeviceError SoundManager::setConfig(const SoundManagerConfig& config) {
     m_pConfig->set(ConfigKey("[Soundcard]","Samplerate"),
                    ConfigValue(static_cast<int>(m_config.getSampleRate())));
 
-    err = setupDevices();
-    if (err == SOUNDDEVICE_ERROR_OK) {
+    status = setupDevices();
```
Comment (ferranpujolcamins) [inline src/soundio/soundmanager.cpp:565]: `status` can just be defined here

**Review by ferranlala — APPROVED** [review]
LGTM, only small nitpick that you didn't even introduce, but would be nice if you could fix.

File: src/soundio/soundmanager.cpp (hunk 1 above)
Comment (ronso0) [reply src/soundio/soundmanager.cpp:565]: Oh sure, I could do that, though I'm touching these files in #4862 anyway, thus I'd rather fix it there and keep this as pure renaming PR.

**Comment by ronso0** [comment]
Anyone else has time for a quick review?

File: src/soundio/sounddevicestatus.h [hunk 2]
```diff
@@ -1,9 +1,9 @@
 #pragma once
 
 // Used for returning errors from sounddevice functions.
-enum SoundDeviceError {
-    SOUNDDEVICE_ERROR_ERR = -1,
-    SOUNDDEVICE_ERROR_OK = 0,
+enum SoundDeviceStatus {
```
Comment (Swiftb0y) [inline src/soundio/sounddevicestatus.h:4]: If you are already touching everything, we might as well make this an enum class IMO. 

```cpp
enum class SoundDeviceStatus {
    Error = -1,
    Ok = 0,
    ErrorDuplicateOutputChannel,
    ErrorExcessiveOutputChannel,
    ErrorExcessiveInputChannel,
    ErrorDeviceCount
};

**Review by Swiftb0y — COMMENTED** [review]
small nitpick if you're in the mood for cleanup. If you think it blows the scope, we can leave as is.

File: src/soundio/sounddevicestatus.h (hunk 2 above)
Comment (ronso0) [reply src/soundio/sounddevicestatus.h]: No prob, done.

File: src/qml/qmlapplication.cpp [hunk 3]
```diff
@@ -47,7 +47,7 @@ QmlApplication::QmlApplication(
           m_fileWatcher({m_mainFilePath}) {
     m_pCoreServices->initialize(app);
     SoundDeviceStatus result = m_pCoreServices->getSoundManager()->setupDevices();
-    if (result != SOUNDDEVICE_OK) {
+    if (result != SoundDeviceStatus::Ok) {
         qCritical() << "Error setting up sound devices" << result;
         exit(result);
```
Comment (Swiftb0y) [inline src/qml/qmlapplication.cpp:52]:
```suggestion
        qCritical() << "Error setting up sound devices" << result;
        exit(static_cast<int>(result));
```

Comment (Swiftb0y) [reply src/qml/qmlapplication.cpp:52]: I think you can fix the `no match for ‘operator<<’ (operand types are ‘QDebug’ and ‘SoundDeviceStatus’)` problem either by just also `static_cast`ing to an int or you can experiment with adding `Q_DECLARE_METATYPE(SoundDeviceStatus)` and/or `
Q_ENUM(SoundDeviceStatus)` in the same file as the enum definition.

Comment (ronso0) [reply src/qml/qmlapplication.cpp:52]: hmm...
For `exit` casting to int is fine.
For the debug output a string would be appreciated. I'm already looking into the latter.
`Q_DECLARE_METATYPE(SoundDeviceStatus);` in src/soundio/sounddevicestatus.h
+
`qDebug() << QVariant::fromValue(status).toString();`
compiles without warnings and works.
Though, is this the preferred solution?

Comment (ronso0) [reply src/qml/qmlapplication.cpp]: I was wrong, to string conversion is tricky / elaborate and I'm not in the mood to crack that nut.
I'll got static_cast<int> for both the debug string and the exit code.

Please double-check, then I'll squash the fixup.

File: src/qml/qmlapplication.cpp [hunk 4]
```diff
@@ -46,10 +46,11 @@ QmlApplication::QmlApplication(
...
     m_pCoreServices->initialize(app);
-    SoundDeviceError result = m_pCoreServices->getSoundManager()->setupDevices();
-    if (result != SOUNDDEVICE_ERROR_OK) {
-        qCritical() << "Error setting up sound devices" << result;
-        exit(result);
+    SoundDeviceStatus result = m_pCoreServices->getSoundManager()->setupDevices();
+    if (result != SoundDeviceStatus::Ok) {
+        int reInt = static_cast<int>(result);
+        qCritical() << "Error setting up sound devices:" << reInt;
+        exit(reInt);
```
Comment (Swiftb0y) [inline src/qml/qmlapplication.cpp:53]: `const` :wink:
```suggestion
        const int reInt = static_cast<int>(result);
        qCritical() << "Error setting up sound devices:" << reInt;
        exit(reInt);
```

**Review by Swiftb0y — APPROVED** [review]
LGTM otherwise.

File: src/qml/qmlapplication.cpp (hunk 4 above)
Comment (ronso0) [reply src/qml/qmlapplication.cpp]: hehe, done

**Comment by Swiftb0y** [comment]
Thanks

**Pull request merged**